app/jobs/update_diaper_partner_job.rb

```diff
@@ -5,16 +5,26 @@ class UpdateDiaperPartnerJob
 
   def perform(partner_id)
     @partner = Partner.find(partner_id)
+    ##NEW CODE FROM HERE DOWN
     @response = nil
-
+    @responseCode = nil
     @response = DiaperPartnerClient.post(@partner.attributes) if Flipper.enabled?(:email_active)
 
+    @responseCode = @response.value
 
-    if @response.value == Net::HTTPSuccess
-    @partner.update(status: "Pending")
-    else
-    @partner.updates(status: "Error")
-    end
+    case @responseCode
+       when Net::HTTPSuccess
+           @partner.update(status: "Pending")
+       when Net::HTTPUnauthorized
+         @partner.update(status: "Error: Unauthorized Access")
+     when Net::HTTPClientError
+       @partner.update(status: "Error: Client Error")
+     when Net::HTTPServerError
+         @partner.update(status: "Error: Server Error")
+       else
+         @partner.update(status: "Error: Unkown Error")
+     end
```

Comment:
This has code smell -- I know I might have been the one to suggest using a case statement, but seeing it now makes me rethink that. 

As far as the app is concerned, we really don't need to exhaustively handle every possible outcome -- all that the app will really care about is "was it successful, or not?" -- so the way you had it originally (the if/else statement) is probably sufficient. (I think the `@partner.updates(status: "Error")` might need to be `@partner.update(status: "Error")` though)

A couple other things (these are Ruby-style things that you may not be aware of):

use underscored-notation for your instance variables, instead of camelcase -- camelcase is used for constants (like Class names), and underscored notation is for variables. so instead of `@responseCode`, you'd use `@response_code`. I'm not 100% sure *why* it's like this, but it is.

A cool thing about ruby is that we don't even need to assign the response code itself to its own variable -- we can just reference the property of the parent object (`@response`), eg. `@response.value`. You can use the `&.` ([safe navigation operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)) instead, like this:

```ruby
if @response&.value == Net::HTTPSuccess
```
or
```ruby
case @response&.value
```

In both cases, what that means is "if `@response` is a non-nil, then send it the `:value` message, but if it's `nil`, then just return `nil` instead". The advantage of this is that the program flow is less disrupted by precautionary setup that might be necessary in other languages. (No worrying about `NullPointerException`s here :) )

If you use the safe-navigation approach, this means you can *also* get rid of `@response = nil` -- and just rely on the result of `DiaperPartnerClient.post(...)...` -- if that fails, it will likely return a `nil` value.

**BONUS**
A common pattern in Ruby is to use fall-through values, by using the `||` operator. So you might have something like:

```ruby
@foo = bar || baz
```
which is equivalent to:
```ruby
if !bar.nil?
  @foo = bar
else
  @foo = baz
end
```
There's even a memoizing operator (`||=`, which, like in many other languages, means: `= _ || _`, like how `+=` means `= _ + _`, which you can use when you want to have a value be provided if a variable is nil. Such as:

```ruby
@foo ||= "This is a default value"
```
which is equivalent to:
```ruby
if @foo.nil?
  @foo = "This is a default value"
end
```

Pulling this all together -- it's common to chain together a series of variables or possible values in order of precedence, and allowing Ruby to use the first non-nil value for assignment.

```ruby
@foo = bar || baz || bat || "default value"
```
This will attempt first to assign the value of `bar` to `@foo`. If `bar` is `nil`, then it will fall through to `baz`, etc., finally settling on `"default value"` if all of the other variables are nil. This is commonly used in Rails with arguments, where you might want to have a default value if the user doesn't provide one.

```ruby
  @partner.name = partner_params[:name] || "No name provided"
```
Which means "Try assigning the value of the `:name` key in the `partner_params` hash, but if it fails to find anything with that key, or if that key exists but has a nil value, then use `"No name provided"`"

or when you want to ensure that *something* is set:

```ruby
  @partner.name ||= "No name provided"
```
Which means: "If `@partner.name` hasn't already been set to something else, then set it to `"No name provided"`"

The use-cases are slightly different there, but it's a similar approach.

Let me know if you'd like to pair up this week!